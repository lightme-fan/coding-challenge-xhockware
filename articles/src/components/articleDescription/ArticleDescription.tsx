import React from 'react'
import styles from "./styles.module.css";

const { detail_container, wrapper, heading, image, title, description, content, author, url, no_read_article } = styles;

const ArticleDescription = ({ article }: any) => {
  return (
    <div className={detail_container}>
      {article       
        ? <div className={wrapper}>
            <div className={heading}>
              <img className={image} src={article?.urlToImage} alt={article?.title} data-testid="image" />
              <div>
                <h2 className={title} data-testid="title">{article?.title}</h2>
                <div data-testid="source">{article?.source?.name}</div>    
              </div>
            </div>
            <div>
              <p className={description} data-testid="description">{article?.description}</p>
              <p className={content} data-testid="content">{article?.content}</p>
              <div className={author} data-testid="author">Writter by {article?.author}</div>
            </div>
            <div className={url}><a href={article?.url} data-testid="article_url">{article?.url}</a></div>
          </div>
        : <div className={no_read_article} data-testid="text_no_data">You haven't open an article yet. Please select one if you want to read!</div>
      }
    </div>
  )
}

export default ArticleDescription