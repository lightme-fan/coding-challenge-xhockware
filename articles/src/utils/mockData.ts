export const article = {
    id: "123",
    source: {
    id: "cnbc",
    name: "CNBC"
  },
    author: "Amelia Lucas",
    title: "Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC",
    description: "Coca-Cola on Friday reported quarterly earnings that met analysts' expectations.",
    url: "https://www.cnbc.com/2019/10/18/coca-cola-ko-earnings-q3-2019.html",
    urlToImage: "https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846",
    publishedAt: "2019-10-18T10:42:46Z",
    content: "Coca-Cola on Friday reported quarterly revenue.",
    viewText: "Read"
}