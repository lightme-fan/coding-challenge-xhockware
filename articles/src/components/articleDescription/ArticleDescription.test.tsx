import '@testing-library/jest-dom'
import ArticleDescription from './ArticleDescription'
import { render } from '@testing-library/react'
import { article } from '../../utils/mockData'

describe("tests the details of an article ", () => {
  test('tests if a text is displayed when there is no article details', async () => {
    const { getByText } = render(<ArticleDescription article={null} />)
    expect(getByText("You haven't open an article yet. Please select one if you want to read!")).toBeInTheDocument()
  })

  test('tests if article details is not null', async () => {
    const component = render(<ArticleDescription article={article} />)
    const { getByTestId } = component;
    expect(getByTestId("title")).toHaveTextContent(article.title);
    expect(getByTestId("author")).toHaveTextContent(article.author);
    expect(getByTestId("description")).toHaveTextContent(article.description);
    expect(getByTestId("content")).toHaveTextContent(article.content);

    expect(component).toMatchSnapshot();
  })
})