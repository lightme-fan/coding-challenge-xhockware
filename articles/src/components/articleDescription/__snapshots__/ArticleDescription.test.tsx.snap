// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the details of an article  tests if article details is not null 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="detail_container"
      >
        <div
          class="wrapper"
        >
          <div
            class="heading"
          >
            <img
              alt="Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC"
              class="image"
              data-testid="image"
              src="https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846"
            />
            <div>
              <h2
                class="title"
                data-testid="title"
              >
                Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC
              </h2>
              <div
                data-testid="source"
              >
                CNBC
              </div>
            </div>
          </div>
          <div>
            <p
              class="description"
              data-testid="description"
            >
              Coca-Cola on Friday reported quarterly earnings that met analysts' expectations.
            </p>
            <p
              class="content"
              data-testid="content"
            >
              Coca-Cola on Friday reported quarterly revenue.
            </p>
            <div
              class="author"
              data-testid="author"
            >
              Writter by 
              Amelia Lucas
            </div>
          </div>
          <div
            class="url"
          >
            <a
              data-testid="article_url"
              href="https://www.cnbc.com/2019/10/18/coca-cola-ko-earnings-q3-2019.html"
            >
              https://www.cnbc.com/2019/10/18/coca-cola-ko-earnings-q3-2019.html
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="detail_container"
    >
      <div
        class="wrapper"
      >
        <div
          class="heading"
        >
          <img
            alt="Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC"
            class="image"
            data-testid="image"
            src="https://image.cnbcfm.com/api/v1/image/105375236-1533317591071rtx6drjw.jpg?v=1571332846"
          />
          <div>
            <h2
              class="title"
              data-testid="title"
            >
              Coca-Cola stock jumps as strong sales of Coke Zero Sugar continue to drive revenue growth - CNBC
            </h2>
            <div
              data-testid="source"
            >
              CNBC
            </div>
          </div>
        </div>
        <div>
          <p
            class="description"
            data-testid="description"
          >
            Coca-Cola on Friday reported quarterly earnings that met analysts' expectations.
          </p>
          <p
            class="content"
            data-testid="content"
          >
            Coca-Cola on Friday reported quarterly revenue.
          </p>
          <div
            class="author"
            data-testid="author"
          >
            Writter by 
            Amelia Lucas
          </div>
        </div>
        <div
          class="url"
        >
          <a
            data-testid="article_url"
            href="https://www.cnbc.com/2019/10/18/coca-cola-ko-earnings-q3-2019.html"
          >
            https://www.cnbc.com/2019/10/18/coca-cola-ko-earnings-q3-2019.html
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
